csrc/cpu/knn_cpu.h

```diff
@@ -0,0 +1,15 @@
+#pragma once
+
+#include <torch/extension.h>
+#include "utils/neighbors.cpp"
+#include <iostream>
+#include "compat.h"
+
+torch::Tensor knn_cpu(torch::Tensor support, torch::Tensor query, 
+			 int64_t k, int64_t n_threads);
+
+torch::Tensor batch_knn_cpu(torch::Tensor support,
+			       torch::Tensor query,
+			       torch::Tensor support_batch,
+			       torch::Tensor query_batch,
+			       int64_t k);
```

Comment:
We can unify this as it is done in the GPU version via `torch::optional`, so that this logic moves from `knn.cpp` to `knn_cpu.cpp`. WDYT?

Reply:
Sounds good to me!

Reply:
I will leave this to you as I'm not sure how exactly you want it done.